routing/dht/providers.go

```diff
@@ -61,17 +61,27 @@ func (pm *ProviderManager) run() {
 			if np.val == pm.lpeer {
 				pm.local[np.k] = struct{}{}
 			}
-			pi := new(providerInfo)
-			pi.Creation = time.Now()
-			pi.Value = np.val
-			arr := pm.providers[np.k]
-			pm.providers[np.k] = append(arr, pi)
+			provs, ok := pm.providers[np.k]
```

Comment:
`pm.run` is the only func that access this map, right? might want to document that in the struct